crates/brioche/src/check.rs

```diff
@@ -1,51 +1,128 @@
-        }
+    // Handle the case where no projects and no registries are specified
+    let projects_path =
+        if args.project.project.is_empty() && args.project.registry_project.is_empty() {
+            vec![PathBuf::from(".")]
+        } else {
+            args.project.project
+        };
 
-        let checked = brioche_core::script::check::check(&brioche, &projects, project_hash).await?;
+    // Loop over the projects
+    for project_path in projects_path {
+        let project_name = format!("project '{name}'", name = project_path.display());
 
-        guard.shutdown_console().await;
+        match projects.load(&brioche, &project_path, true).await {
+            Ok(project_hash) => {
+                let result =
+                    run_check(&reporter, &brioche, &projects, project_hash, &project_name).await;
+                consolidate_result(&reporter, &project_name, result, &mut error_result);
+            }
+            Err(e) => {
+                consolidate_result(&reporter, &project_name, Err(e), &mut error_result);
+            }
+        }
+    }
 
-        let result = checked.ensure_ok(brioche_core::script::check::DiagnosticLevel::Message);
+    // Loop over the registry projects
+    for registry_project in args.project.registry_project {
+        let project_name = format!("registry project '{registry_project}'");
 
-        match result {
-            Ok(()) => {
-                println!("No errors found 🎉");
-                anyhow::Ok(ExitCode::SUCCESS)
+        match projects
+            .load_from_registry(
+                &brioche,
+                &registry_project,
+                &brioche_core::project::Version::Any,
+            )
+            .await
+        {
+            Ok(project_hash) => {
+                let result =
+                    run_check(&reporter, &brioche, &projects, project_hash, &project_name).await;
+                consolidate_result(&reporter, &project_name, result, &mut error_result);
             }
-            Err(diagnostics) => {
-                diagnostics.write(&brioche.vfs, &mut std::io::stdout())?;
-                anyhow::Ok(ExitCode::FAILURE)
+            Err(e) => {
+                consolidate_result(&reporter, &project_name, Err(e), &mut error_result);
             }
         }
-    };
+    }
 
-    let exit_code = check_future
-        .instrument(tracing::info_span!("check"))
-        .await?;
+    guard.shutdown_console().await;
+
+    let exit_code = if error_result.is_some() {
+        ExitCode::FAILURE
+    } else {
+        ExitCode::SUCCESS
+    };
 
     Ok(exit_code)
 }
+
+async fn run_check(
+    reporter: &Reporter,
+    brioche: &Brioche,
+    projects: &Projects,
+    project_hash: ProjectHash,
+    project_name: &String,
+) -> Result<bool, anyhow::Error> {
+    let num_lockfiles_updated = projects.commit_dirty_lockfiles().await?;
+    if num_lockfiles_updated > 0 {
+        tracing::info!(num_lockfiles_updated, "updated lockfiles");
```

Comment:
One thing, I didn't touch this line, but should I?

Reply:
I think this is fine as-is, is there a reason it should be changed?

Reply:
No, I don't think so. I just wanted to raise the attention to this single call to `tracing::info`.